cdap-watchdog/src/main/java/io/cdap/cdap/logging/pipeline/queue/TimeEventQueueProcessor.java

```diff
@@ -127,6 +127,13 @@ private ProcessedEventMetadata<OFFSET> append() {
       minDelay = delay < minDelay ? delay : minDelay;
       maxDelay = delay > maxDelay ? delay : maxDelay;
 
+      // Skip entry if the logger name is null. This avoids an NPE when appending to the effectiveLogger below.
+      if (event.getLoggerName() == null) {
```

Comment:
any other assumptions we should be checking for?

may also be good to add some logic to have a max number or retries. As we're only dealing with log messages it should be ok to drop a problematic one that causes multiple exceptions in a row.